### ISSUE TITLE
Error from clientFilesQuestion

### ISSUE THREAD

**Issue by mwest1066** [PrairieLearn/PrairieLearn#3181]
We had multiple servers trigger this error. The request that triggered it was:
```
{"timestamp":"2020-09-23T13:10:43.067Z","ip":"172.58.236.225","forwardedIP":"REDACTED","authn_user_id":"REDACTED","authn_user_uid":"REDACTED","method":"GET","path":"/pl/course_instance/128344/instructor/question/9047433/clientFilesQuestion/","params":{},"body":{},"response_id":"fd557be5-5e28-455e-b636-35337be14868","level":"verbose","message":"request"}
```

The error log was:
```
Sep 23 13:10:43 ip-10-0-0-204.us-east-2.compute.internal node[120113]: /PrairieLearn/node_modules/express/lib/response.js:411
Sep 23 13:10:43 ip-10-0-0-204.us-east-2.compute.internal node[120113]: throw new TypeError('path argument is required to res.sendFile');
Sep 23 13:10:43 ip-10-0-0-204.us-east-2.compute.internal node[120113]: ^
Sep 23 13:10:43 ip-10-0-0-204.us-east-2.compute.internal node[120113]: TypeError: path argument is required to res.sendFile
Sep 23 13:10:43 ip-10-0-0-204.us-east-2.compute.internal node[120113]: at ServerResponse.sendFile (/PrairieLearn/node_modules/express/lib/response.js:411:11)
Sep 23 13:10:43 ip-10-0-0-204.us-east-2.compute.internal node[120113]: at Object.<anonymous> (/PrairieLearn/pages/clientFilesQuestion/clientFilesQuestion.js:24:13)
Sep 23 13:10:43 ip-10-0-0-204.us-east-2.compute.internal node[120113]: at cb (util.js:207:31)
Sep 23 13:10:43 ip-10-0-0-204.us-east-2.compute.internal node[120113]: at processTicksAndRejections (internal/process/task_queues.js:84:21)
```

**Comment by mwest1066** [comment]
From @nicknytko: maybe the `filename` was undefined here: https://github.com/PrairieLearn/PrairieLearn/blob/b7c3105ccd65bbb6316ccf5bf13cac26f235cc68/pages/clientFilesQuestion/clientFilesQuestion.js#L24

From the above request log it looks like the route was `/pl/course_instance/128344/instructor/question/9047433/clientFilesQuestion/` which is indeed missing the filename on the end of the URL: https://github.com/PrairieLearn/PrairieLearn/blob/b7c3105ccd65bbb6316ccf5bf13cac26f235cc68/pages/clientFilesQuestion/clientFilesQuestion.js#L9

I guess we should return a 400 if the filename is missing?

**Comment by mwest1066** [comment]
There might be other places in the code where we do a similar thing to get a filename, which could cause the same error. We should check for these.

Alternatively, if we convert to using `asyncHandler` for our routes then at least we wouldn't crash with this type of unhandled exception. E.g., https://github.com/PrairieLearn/PrairieLearn/blob/b7c3105ccd65bbb6316ccf5bf13cac26f235cc68/pages/administratorQueries/administratorQueries.js#L12

**Comment by nicknytko** [comment]
I see that we have this decodePath (https://github.com/PrairieLearn/PrairieLearn/blob/0019d997bf0d7e42588faa94130da23e4d84f888/lib/uri-util.js#L17) helper that we use in some places to parse url paths.  You think we should use it here?  It looks to correctly handle null or undefined values.

**Comment by mwest1066** [comment]
I don't think we need all the path-decoding code in `decodePath()`? Maybe just a simple check for `if (filename == null)` would be sufficient here?

**Comment by nicknytko** [comment]
> I don't think we need all the path-decoding code in `decodePath()`? Maybe just a simple check for `if (filename == null)` would be sufficient here?

Yep, totally fair.  Just wondering since I see we're already doing it in some places before we send a file.

**Comment by mwest1066** [comment]
Same as #3349 — fixed by #3474
